File: 프로그래머스/1/68644. 두 개 뽑아서 더하기/두 개 뽑아서 더하기.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>
using namespace std;

vector<int> solution(vector<int> numbers) {
    vector<int> answer;
    
    for(int i=0; i<numbers.size()-1; i++) {
        for(int j=i+1; j<numbers.size(); j++) {
            int total = numbers[i] + numbers[j];
            
            if(find(answer.begin(), answer.end(), total) == answer.end()) {
                answer.push_back(total);    
            }
        }
    }
    
    sort(answer.begin(), answer.end());
    
    return answer;
}

File: 프로그래머스/1/68644. 두 개 뽑아서 더하기/두 개 뽑아서 더하기.java
import java.util.*;
class Solution {
    public int[] solution(int[] numbers) {
        Set<Integer> set = new HashSet<>();
        
        for(int i=0; i<numbers.length; i++) {
            for(int j=i+1; j<numbers.length; j++) {
                int sum = numbers[i] + numbers[j];
                set.add(sum);
            }
        }
        
        int[] res = new int[set.size()];
        int i = 0;
        for(int x : set) {
            res[i++] = x;
        }
        
        Arrays.sort(res);
        
        return res;
    }
}